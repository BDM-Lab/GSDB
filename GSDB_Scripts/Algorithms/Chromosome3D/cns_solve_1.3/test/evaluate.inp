remarks EVALUATE
remarks test case for new EVALUATE commands
remarks by Mark McCallum Yale, 23-MAY-88

remarks attempt simple assignments (assume min of 9 variables)

evaluate ( $1 = -4 )
evaluate ( $2 = -3 )
evaluate ( $3 = -2 )
evaluate ( $4 = -1 )
evaluate ( $5 = 0 )
evaluate ( $6 = 1 )
evaluate ( $7 = 2 )
evaluate ( $8 = 3 )
evaluate ( $9 = 4 )

remarks perform slightly more complex expressions

evaluate ( $1 = 10 / 5 + 3 - 2 * 4 )
evaluate ( $2 = 25 ** .5 )
evaluate ( $3 = 3 * ( 2 + 1 ) / ( 5 - 3) )
evaluate ( $4 = ( 3 + 2 ) * ( 7 * ( 3 - 1 ) ) )
evaluate ( $5 = ~3 )
evaluate ( $6 = 4 - -4 + 3 / 4 * 3 - ~1 + 2^2 )
evaluate ( $7 = 2^2^3**4 )

remarks include variables on right hand side of expression

evaluate ( $1 = 10 / $1 + 3 - $2 * $3 )
evaluate ( $2 = $1 * $2 / $3 )
evaluate ( $3 = $2 * ( $1 + $2 ) / ( $3 - 3) )
evaluate ( $4 = ( $3 + $1 ) * ( $3 * ( $3 - 1 ) ) )
evaluate ( $5 = ~$3 )
evaluate ( $6 = $4 - -$3 + 3 / 4 * $6 - ~1 + $1^2 )
evaluate ( $7 = ( $1* -.1 )^2^3**($3 /100) )

remarks include functions (19 of them)

evaluate ( $1 = max( $1, .33, 33.3, .7, 3 ) )
evaluate ( $1 = min( $1, .33, 33.3, .7, 3 ) )
evaluate ( $1 = sqrt( $1 ) )
evaluate ( $1 = log10( $1 ) )
evaluate ( $1 = abs($1) )
evaluate ( $1 = log( $1 ) )
evaluate ( $1 = heavy( $1 ) )
evaluate ( $1 = cos( $1 ) )
evaluate ( $1 = sin( $1 ) )
evaluate ( $1 = asin( 0.4 ) )
evaluate ( $1 = acos( 0.5 ) )
evaluate ( $1 = int( $1 ) )
evaluate ( $1 = tan( $1 ) )
evaluate ( $1 = exp( $1 ) )
evaluate ( $1 = mod( ($1+5), 2 ) )
evaluate ( $1 = encode( $1 ) )
evaluate ( $1 = decode( $1 ) )
evaluate ( $1 = gaussian( $1 ) )
evaluate ( $1 = sign( $1 ) )
evaluate ( $1 =random() )

evaluate ( $2 = "abCDefG" )
evaluate ( $3 = capitalize ($2 ) ) 
evaluate ( $3 = capitalize("abCDefG" ) ) 


evaluate ( $1 = "rr"+")" )
evaluate ( $1 = "rr"+"+" +"(" )
evaluate ( $1 = "rr"+"-" )
evaluate ( $1 = "rr"+"*" )

evaluate ( $1 = random()-4.0 )
evaluate ( $1 = exp(-3.) )

evaluate ( $1 = imod( 3.1,1.9 ) )
evaluate ( $1= nint(3.1) )
evaluate ( $1= nint(2.9) ) 
evaluate ( $1= nint(-3.1) )
evaluate ( $1= nint(-2.9) ) 
evaluate ( $1 = imod( -3.1,-1.9 ) )



evaluate ($1=TRUE)
evaluate ($2=true)
evaluate ($3="true")
evaluate ($4="TRUE")
evaluate ($5=false)
evaluate ($6=FALSE)

xray
  anomalous=$1
  anomalous ?
  evaluate ($10=$result)
  declare name=fobs type=complex domain=reciprocal end
  
   refl 
    index  1 2 3 fobs 50 -50
  end

  do (fobs=fobs*2) ( $1 and amplitude(fobs)>0) 
  do (fobs=fobs*8) ( $10 and amplitude(fobs)>0)
  do (fobs=fobs*4) ( $5 and amplitude(fobs)>0)
end
display  $1  $2  $3  $4 $5 $6 $10   
if ($10=TRUE) then
end if

if ($1=false) then
end if

if ($10=true) then
end if

if ($1="TRUE" ) then
end if

stop 
