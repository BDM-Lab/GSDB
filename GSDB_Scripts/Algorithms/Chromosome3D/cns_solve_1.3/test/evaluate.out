          ============================================================
          |                                                          |
          |            Crystallography & NMR System (CNS)            |
          |                         CNSsolve                         |
          |                                                          |
          ============================================================
           Version: 1.3
           Status: General release
          ============================================================
           Written by: A.T.Brunger, P.D.Adams, G.M.Clore, W.L.DeLano,
                       P.Gros, R.W.Grosse-Kunstleve,J.-S.Jiang,J.M.Krahn,
                       J.Kuszewski, M.Nilges, N.S.Pannu, R.J.Read,
                       L.M.Rice, G.F.Schroeder, T.Simonson, G.L.Warren.
           Copyright (c) 1997-2010 Yale University
          ============================================================
           Running on machine: lewis4-r710-login-node223 (x86_64/Linux,64-bit)
                               with 16 threads
           Program started by: oeow39
           Program started at: 16:08:00 on 05-Jul-2018
          ============================================================

 FFT3C: Using FFTPACK4.1

 CNSsolve>remarks EVALUATE 
 CNSsolve>remarks test case for new EVALUATE commands 
 CNSsolve>remarks by Mark McCallum Yale, 23-MAY-88 
 CNSsolve> 
 CNSsolve>remarks attempt simple assignments (assume min of 9 variables) 
 CNSsolve> 
 CNSsolve>evaluate ( $1 = -4 ) 
 EVALUATE: symbol $1 set to   -4.00000     (real)
 CNSsolve>evaluate ( $2 = -3 ) 
 EVALUATE: symbol $2 set to   -3.00000     (real)
 CNSsolve>evaluate ( $3 = -2 ) 
 EVALUATE: symbol $3 set to   -2.00000     (real)
 CNSsolve>evaluate ( $4 = -1 ) 
 EVALUATE: symbol $4 set to   -1.00000     (real)
 CNSsolve>evaluate ( $5 = 0 ) 
 EVALUATE: symbol $5 set to    0.00000     (real)
 CNSsolve>evaluate ( $6 = 1 ) 
 EVALUATE: symbol $6 set to    1.00000     (real)
 CNSsolve>evaluate ( $7 = 2 ) 
 EVALUATE: symbol $7 set to    2.00000     (real)
 CNSsolve>evaluate ( $8 = 3 ) 
 EVALUATE: symbol $8 set to    3.00000     (real)
 CNSsolve>evaluate ( $9 = 4 ) 
 EVALUATE: symbol $9 set to    4.00000     (real)
 CNSsolve> 
 CNSsolve>remarks perform slightly more complex expressions 
 CNSsolve> 
 CNSsolve>evaluate ( $1 = 10 / 5 + 3 - 2 * 4 ) 
 EVALUATE: symbol $1 set to   -3.00000     (real)
 CNSsolve>evaluate ( $2 = 25 ** .5 ) 
 EVALUATE: symbol $2 set to    5.00000     (real)
 CNSsolve>evaluate ( $3 = 3 * ( 2 + 1 ) / ( 5 - 3) ) 
 EVALUATE: symbol $3 set to    4.50000     (real)
 CNSsolve>evaluate ( $4 = ( 3 + 2 ) * ( 7 * ( 3 - 1 ) ) ) 
 EVALUATE: symbol $4 set to    70.0000     (real)
 CNSsolve>evaluate ( $5 = ~3 ) 
 EVALUATE: symbol $5 set to   -3.00000     (real)
 CNSsolve>evaluate ( $6 = 4 - -4 + 3 / 4 * 3 - ~1 + 2^2 ) 
 EVALUATE: symbol $6 set to    15.2500     (real)
 CNSsolve>evaluate ( $7 = 2^2^3**4 ) 
 EVALUATE: symbol $7 set to   0.167772E+08 (real)
 CNSsolve> 
 CNSsolve>remarks include variables on right hand side of expression 
 CNSsolve> 
 CNSsolve>evaluate ( $1 = 10 / $1 + 3 - $2 * $3 ) 
 EVALUATE: symbol $1 set to   -22.8333     (real)
 CNSsolve>evaluate ( $2 = $1 * $2 / $3 ) 
 EVALUATE: symbol $2 set to   -25.3704     (real)
 CNSsolve>evaluate ( $3 = $2 * ( $1 + $2 ) / ( $3 - 3) ) 
 EVALUATE: symbol $3 set to    815.297     (real)
 CNSsolve>evaluate ( $4 = ( $3 + $1 ) * ( $3 * ( $3 - 1 ) ) ) 
 EVALUATE: symbol $4 set to   0.526112E+09 (real)
 CNSsolve>evaluate ( $5 = ~$3 ) 
 EVALUATE: symbol $5 set to   -815.297     (real)
 CNSsolve>evaluate ( $6 = $4 - -$3 + 3 / 4 * $6 - ~1 + $1^2 ) 
 EVALUATE: symbol $6 set to   0.526114E+09 (real)
 CNSsolve>evaluate ( $7 = ( $1* -.1 )^2^3**($3 /100) ) 
 EVALUATE: symbol $7 set to   0.347083E+18 (real)
 CNSsolve> 
 CNSsolve>remarks include functions (19 of them) 
 CNSsolve> 
 CNSsolve>evaluate ( $1 = max( $1, .33, 33.3, .7, 3 ) ) 
 EVALUATE: symbol $1 set to    33.3000     (real)
 CNSsolve>evaluate ( $1 = min( $1, .33, 33.3, .7, 3 ) ) 
 EVALUATE: symbol $1 set to   0.330000     (real)
 CNSsolve>evaluate ( $1 = sqrt( $1 ) ) 
 EVALUATE: symbol $1 set to   0.574456     (real)
 CNSsolve>evaluate ( $1 = log10( $1 ) ) 
 EVALUATE: symbol $1 set to  -0.240743     (real)
 CNSsolve>evaluate ( $1 = abs($1) ) 
 EVALUATE: symbol $1 set to   0.240743     (real)
 CNSsolve>evaluate ( $1 = log( $1 ) ) 
 EVALUATE: symbol $1 set to   -1.42403     (real)
 CNSsolve>evaluate ( $1 = heavy( $1 ) ) 
 EVALUATE: symbol $1 set to    0.00000     (real)
 CNSsolve>evaluate ( $1 = cos( $1 ) ) 
 EVALUATE: symbol $1 set to    1.00000     (real)
 CNSsolve>evaluate ( $1 = sin( $1 ) ) 
 EVALUATE: symbol $1 set to   0.174524E-01 (real)
 CNSsolve>evaluate ( $1 = asin( 0.4 ) ) 
 EVALUATE: symbol $1 set to    23.5782     (real)
 CNSsolve>evaluate ( $1 = acos( 0.5 ) ) 
 EVALUATE: symbol $1 set to    60.0000     (real)
 CNSsolve>evaluate ( $1 = int( $1 ) ) 
 EVALUATE: symbol $1 set to    59.0000     (real)
 CNSsolve>evaluate ( $1 = tan( $1 ) ) 
 EVALUATE: symbol $1 set to    1.66428     (real)
 CNSsolve>evaluate ( $1 = exp( $1 ) ) 
 EVALUATE: symbol $1 set to    5.28187     (real)
 CNSsolve>evaluate ( $1 = mod( ($1+5), 2 ) ) 
 EVALUATE: symbol $1 set to   0.281866     (real)
 CNSsolve>evaluate ( $1 = encode( $1 ) ) 
 EVALUATE: symbol $1 set to "0.281866" (string)
 CNSsolve>evaluate ( $1 = decode( $1 ) ) 
 EVALUATE: symbol $1 set to   0.281866     (real)
 CNSsolve>evaluate ( $1 = gaussian( $1 ) ) 
 EVALUATE: symbol $1 set to  -0.193952     (real)
 CNSsolve>evaluate ( $1 = sign( $1 ) ) 
 EVALUATE: symbol $1 set to   -1.00000     (real)
 CNSsolve>evaluate ( $1 =random() ) 
 EVALUATE: symbol $1 set to   0.242169     (real)
 CNSsolve> 
 CNSsolve>evaluate ( $2 = "abCDefG" ) 
 EVALUATE: symbol $2 set to "abCDefG" (string)
 CNSsolve>evaluate ( $3 = capitalize ($2 ) ) 
 EVALUATE: symbol $3 set to "ABCDEFG" (string)
 CNSsolve>evaluate ( $3 = capitalize("abCDefG" ) ) 
 EVALUATE: symbol $3 set to "ABCDEFG" (string)
 CNSsolve> 
 CNSsolve> 
 CNSsolve>evaluate ( $1 = "rr"+")" ) 
 EVALUATE: symbol $1 set to "rr)" (string)
 CNSsolve>evaluate ( $1 = "rr"+"+" +"(" ) 
 EVALUATE: symbol $1 set to "rr+(" (string)
 CNSsolve>evaluate ( $1 = "rr"+"-" ) 
 EVALUATE: symbol $1 set to "rr-" (string)
 CNSsolve>evaluate ( $1 = "rr"+"*" ) 
 EVALUATE: symbol $1 set to "rr*" (string)
 CNSsolve> 
 CNSsolve>evaluate ( $1 = random()-4.0 ) 
 EVALUATE: symbol $1 set to   -3.86033     (real)
 CNSsolve>evaluate ( $1 = exp(-3.) ) 
 EVALUATE: symbol $1 set to   0.497871E-01 (real)
 CNSsolve> 
 CNSsolve>evaluate ( $1 = imod( 3.1,1.9 ) ) 
 EVALUATE: symbol $1 set to    1.00000     (real)
 CNSsolve>evaluate ( $1= nint(3.1) ) 
 EVALUATE: symbol $1 set to    3.00000     (real)
 CNSsolve>evaluate ( $1= nint(2.9) ) 
 EVALUATE: symbol $1 set to    3.00000     (real)
 CNSsolve>evaluate ( $1= nint(-3.1) ) 
 EVALUATE: symbol $1 set to   -3.00000     (real)
 CNSsolve>evaluate ( $1= nint(-2.9) ) 
 EVALUATE: symbol $1 set to   -3.00000     (real)
 CNSsolve>evaluate ( $1 = imod( -3.1,-1.9 ) ) 
 EVALUATE: symbol $1 set to   -1.00000     (real)
 CNSsolve> 
 CNSsolve> 
 CNSsolve> 
 CNSsolve>evaluate ($1=TRUE) 
 EVALUATE: symbol $1 set to TRUE (logical)
 CNSsolve>evaluate ($2=true) 
 EVALUATE: symbol $2 set to TRUE (logical)
 CNSsolve>evaluate ($3="true") 
 EVALUATE: symbol $3 set to "true" (string)
 CNSsolve>evaluate ($4="TRUE") 
 EVALUATE: symbol $4 set to "TRUE" (string)
 CNSsolve>evaluate ($5=false) 
 EVALUATE: symbol $5 set to FALSE (logical)
 CNSsolve>evaluate ($6=FALSE) 
 EVALUATE: symbol $6 set to FALSE (logical)
 CNSsolve> 
 CNSsolve>xray 
 XRAY>  anomalous=$1 
 XRAY>  anomalous ? 
 ANOMalous=TRUE {ON}
 XRAY>  evaluate ($10=$result) 
 EVALUATE: symbol $10 set to TRUE (logical)
 XRAY>  declare name=fobs type=complex domain=reciprocal end 
 XDECLARE: Object FOBS has been declared.
 XRAY> 
 XRAY>   refl 
 REFLection>    index  1 2 3 fobs 50 -50 
 XSFAL: allocating space for complex reciprocal space object.
 XRRR2:        1 new h,k,l indices have been added.
 XRAY> 
 XRAY>  do (fobs=fobs*2) ( $1 and amplitude(fobs)>0) 
 Total of         1 structure factor elements were selected.
 XRAY>  do (fobs=fobs*8) ( $10 and amplitude(fobs)>0) 
 Total of         1 structure factor elements were selected.
 XRAY>  do (fobs=fobs*4) ( $5 and amplitude(fobs)>0) 
 Total of         0 structure factor elements were selected.
 XRAY>end 
 CNSsolve>display  $1  $2  $3  $4 $5 $6 $10 
 TRUE  TRUE  true  TRUE FALSE FALSE TRUE
 CNSsolve>if ($10=TRUE) then 
 NEXTCD: condition evaluated as true
 CNSsolve>end if 
 CNSsolve> 
 CNSsolve>if ($1=false) then 
 NEXTCD: condition evaluated as false
 CNSsolve>end if 
 CNSsolve> 
 CNSsolve>if ($10=true) then 
 NEXTCD: condition evaluated as true
 CNSsolve>end if 
 CNSsolve> 
 CNSsolve>if ($1="TRUE" ) then 
 %NEXTCD-ERR: Mismatched data types:
 if ($1="TRUE" 
        ^^^^^^
 NEXTCD: condition evaluated as false
 CNSsolve>end if 
 CNSsolve> 
 CNSsolve>stop 
          ============================================================
           Maximum dynamic memory allocation:     8201488 bytes
           Maximum dynamic memory overhead:           416 bytes
           Program started at: 16:08:00 on 05-Jul-2018
           Program stopped at: 16:08:00 on 05-Jul-2018
           CPU time used:       0.0065 seconds
          ============================================================
